### ISSUE TITLE
[Bug]: node-patches readlink/readlinkSync computes wrong absolute path (joins target onto unresolved parent dir)

### ISSUE THREAD

**Issue by mloeppky-zoox** [aspect-build/rules_js#2914]
### What happened?

The Node `fs` monkey-patches in `js/private/node-patches/fs.cjs` compute the wrong absolute path for a symlink target in `readlink` and `readlinkSync`. They join the raw kernel link target onto the **lexical** parent directory of the *queried* path:

```js
// js/private/node-patches/fs.cjs — readlinkSync (and the async readlink callback)
const resolved = resolvePathLike(args[0]);
const str = path.resolve(path.dirname(resolved), origReadlinkSync(...args));
```

`path.dirname(resolved)` is the parent of the path *as the caller wrote it*, which may itself traverse other symlinks. When the caller reaches the link through symlinks whose resolution changes the directory depth, the relative (`../…`) link target is counted from the wrong base and the resulting absolute path lands one directory level too high. Because that wrong path is still *inside* the guard root, no escape is detected and the bogus path is handed back to the caller.

Any tool that resolves modules via `readlink` (here: eslint-plugin-import's `import/namespace` rule through `eslint-import-resolver-typescript` → `enhanced-resolve`) then `stat`s a path that doesn't exist and fails with `ENOENT`.

Concretely, resolving a first-party package reached through an aliased package-store path that is one path-segment shallower than the real package directory produced an absolute path missing an intermediate segment, e.g.:

```
Error: ENOENT: no such file or directory, stat
  '.../bin/<pkg>/lint_/lint.runfiles/node_modules/.aspect_rules_js/@myorg+web@0.0.0/node_modules/@myorg/base/dist/esm/subpath/index.js'
Occurred while linting .../src/foo.ts:1
Rule: "import/namespace"
```

**Expected:** `readlink`/`readlinkSync` should return a target resolved against the *real* location of the link's parent directory, so the absolute path is correct regardless of how many symlinks the caller path traversed.

### Version

Development (host) and target OS/architectures: Linux x86_64 (host + target)

Output of `bazel --version`: `bazel 7.7.1-1`

Version of the Aspect rules: `rules_js` v2.8.2 (via a vendored archive). **The buggy code is unchanged on `main` / v3.2.2** — `readlink`/`readlinkSync` in `fs.cjs` still use `path.resolve(path.dirname(resolved), …)` with the unresolved parent — so this is not fixed in the latest release.

Language(s) and/or frameworks involved: TypeScript, Node.js v22, ESLint (`eslint-plugin-import` `import/namespace` + `eslint-import-resolver-typescript`), pnpm workspaces, first-party `js_library` packages linked via `npm_link_all_packages`.

### How to reproduce

```shell
# Shape of the repro (in a rules_js pnpm-workspace monorepo):
# - first-party package @myorg/web (js_library) that re-exports from @myorg/base
# - @myorg/base has a package.json "types"/"exports" that points into ./dist
#   (so a resolver follows the dist entry, which re-exports a subpath)
# - an eslint_test on a consumer that imports @myorg/web
#
# The consumer resolves @myorg/base through the package-store alias
#   node_modules/.aspect_rules_js/@myorg+web@0.0.0/node_modules/@myorg/base
# which is reached via symlinks of a different depth than the real
# bazel-out package dir, so readlink's lexical-parent join is off by one level.

bazel test //path/to/consumer:lint   # eslint_test → ENOENT under import/namespace

# Minimal isolation (no bazel): load js/private/node-patches/register.cjs with
# JS_BINARY__FS_PATCH_ROOTS set, then fs.readlinkSync() a symlink reached through
# a parent symlink of different depth — the returned absolute path is wrong.
# Realpath-ing the parent before joining fixes it.
```

### Any other information?

Root cause is purely the lexical-vs-real parent in the two `readlink` implementations. Suggested fix (identity transform whenever the caller path is already fully resolved, so it only changes results that were previously wrong):

```js
let linkDir = path.dirname(resolved);
try { linkDir = origRealpathSyncNative(linkDir); } catch (_) {}
const str = path.resolve(linkDir, origReadlinkSync(...args));
```

Apply the same to the async `readlink` callback. Cost is one guarded `realpath` syscall per `readlink`.
